v0/destinations/klaviyo/util.js

```diff
@@ -0,0 +1,175 @@
+const get = require("get-value");
+const { httpGET } = require("../../../adapters/network");
+const logger = require("../../../logger");
+
+const { WhiteListedTraits } = require("../../../constants");
+
+const {
+  defaultRequestConfig,
+  constructPayload,
+  getFieldValueFromMessage,
+  isDefinedAndNotNull,
+  removeUndefinedValues,
+  defaultPostRequestConfig,
+  extractCustomFields,
+  removeUndefinedAndNullValues
+} = require("../../util");
+
+const {
+  BASE_ENDPOINT,
+  LIST_CONF,
+  MAPPING_CONFIG,
+  CONFIG_CATEGORIES
+} = require("./config");
+
+/**
+ * This function is used to check if the user/profile already exists or not, if already exists unique person_id 
+ * for that user is getting returned else false is returned.
+ * Docs: https://developers.klaviyo.com/en/reference/get-profile-id
+ * @param {*} message 
+ * @param {*} Config 
+ * @returns
+ */
+const isProfileExist = async (message, { Config }) => {
+  const { privateApiKey } = Config;
+  const userIdentifiers = {
+    email: getFieldValueFromMessage(message, "email"),
+    external_id: getFieldValueFromMessage(message, "userId"),
+    phone_number: getFieldValueFromMessage(message, "phone")
+  };
+  
+  for (const id in userIdentifiers) {
+    if (isDefinedAndNotNull(userIdentifiers[id])) {
+      const profileResponse = await httpGET(
+        `${BASE_ENDPOINT}/api/v2/people/search`,
+        {
+          header: {
+            Accept: "application/json"
+          },
+          params: {
+            api_key: privateApiKey,
+            [id]: userIdentifiers[id]
+          }
+        }
+      );
+      if (profileResponse.success && profileResponse.response?.data?.id) {
```

Comment:
We are not handling error scenario i.e `profileResponse.success = false`. We should respond back to server with proper error in case of request failure. Currently in case of failures we are by default returning false but in that case we are making a wrong assumption that profile is not exisiting but we don't know actually as we were unable to make a sucessfull request

Reply:
Means for false success response, we should throw error. Right?

Reply:
Yes also in error we should send whats the response from after the request failed. You can use this function to parse the response to dynamically understand status, message and then wrap those by throwing error

Reply:
@ujjwal-ab this is a critical point